client.go

```diff
@@ -554,23 +554,36 @@ func (client *client) update(data *MetadataResponse) ([]string, error) {
 
 	var err error
 	for _, topic := range data.Topics {
+
+		delete(client.metadata, topic.Name)
+		delete(client.cachedPartitionsResults, topic.Name)
+
 		switch topic.Err {
 		case ErrNoError:
 			break
-		case ErrLeaderNotAvailable, ErrUnknownTopicOrPartition:
+		case ErrInvalidTopic: // don't retry, don't store partial results
+			err = topic.Err
```

Comment:
which is fine, I think - we only get this if you explicitly pass in "SomeInvalidTopic\\\\\" to `RefreshMetadata`, so returning `ErrInvalidTopic` in that case is correct, even if not all topics were invalid